Add Curated Module v2 guide to /run-on-lido (#881)
lidofinance/docs: run-on-lido/cm-v2/bond-and-key-management.md

---
sidebar_position: 6
---

# 🔐 Bond & Key Management

This page covers how to upload and manage your validator keys, and how bond works in CMv2. Bond is collateral you post to back your keys. The amount required depends on your Node Operator type and number of keys.

---

## Key management

### Uploading keys

![Upload keys](/img/cm-guide/bond-upload-keys.png)

To upload keys, go to the keys tab in cm.lido.fi, and drag and drop the `deposit-data.json` file from your keys. The widget will tell you how much bond is required for the amount of keys submitted.

### How deposits work

> After valid keys are uploaded, they become eligible for deposits. Deposits in CMv2 happen in two stages:

Reviewer: We need a short note on what to do with invalid keys and what keys are considered invalid. If we already have, just link it here

Reviewer: Still relevant

---
sidebar_position: 6
---

# 🔐 Bond & Key Management

> This page covers how to upload and manage your validator keys, and how bond works in CMv2. Bond is collateral you post to back your keys. The amount required depends on your Node Operator type and number of keys.

Reviewer 2: ```suggestion
This page covers how to upload and manage your validator keys, and how bond works in CMv2. Bond is collateral you post to activate new validators from keys provided and back them. The amount required depends on your Node Operator type and number of keys.
```

---

## Key management

### Uploading keys

![Upload keys](/img/cm-guide/bond-upload-keys.png)

> To upload keys, go to the keys tab in [cm.lido.fi](https://cm.lido.fi), and drag and drop the `deposit-data.json` file from your keys. The widget will tell you how much bond is required for the number of keys submitted.

Reviewer 2: how to obtain `deposit-data.json` tho?

Reviewer: This is a guide for PROS! But we can add it for sure

> Your keys will be considered invalid if they have an invalid signature, or if they have been uploaded to Lido or already used on the Beacon Chain. In that case, you’ll have to remove the invalid key(s) and resubmit them correctly.

Reviewer 2: Intro paragraph says keys are "considered invalid if they have an invalid signature, or if they have been uploaded to Lido or already used on the Beacon Chain", but the status table immediately below distinguishes Invalid (bad signature) from Duplicated (already uploaded / used on CL). Tighten the intro so the two failure modes are not labeled the same way.

The [Key statuses](#key-statuses) section below provides more information on all the possible states of your keys.

### How deposits work

After valid keys are uploaded, they become eligible for deposits. Deposits in CMv2 happen in two stages:

- **Initial deposit:** a validator first receives its initial 32 ETH deposit to get activated.
- **Top-ups:** after activation, the same validator can receive additional deposits over time.

> CMv2 does not deposit validators in simple upload order. New `0x02` keys are deposited 32 ETH to get activated and then the protocol will deposit into them until they reach 2,048 ETH.

Reviewer 2: It will not necessarily — top-ups depend on the weighted allocation, target stake, deposit queue depth, and the operator's relative position. Should be "may receive top-ups up to 2,048 ETH" (the next paragraph correctly hedges; the first sentence undercuts it).

> These top-ups are determined by the module's weighted allocation algorithm, which prioritizes Node Operators that are furthest below their target stake allocation.

Reviewer 2: and seeding initial deposits before top-ups

| **Active** | Key is active on the beacon chain | Make sure your validator node is online to perform its duties |
| **Exited** | Key has been exited | — |
| **Withdrawn** | Key has been exited and ETH has been returned to the protocol | — |
| **Unbonded** | Bond is insufficient for this key, which can be Active or otherwise | Top up bond or remove/exit the key |
| **Exit requested** | An exit has been requested by [VEBO](https://docs.lido.fi/contracts/validators-exit-bus-oracle/) but the validator has not yet exited | Exit the validator as soon as possible to avoid a Late exit penalty |
| **Duplicated** | Key has been uploaded twice either to the Lido protocol or Ethereum CL | Remove duplicate key |
| **Invalid** | Uploaded key has an invalid signature | Remove key and re-upload it with the valid signature |

### Exiting keys

![Exit keys](/img/cm-guide/bond-exit-keys.png)

Operators in the Curated Module can exit their keys through the following options:

#### Removing keys before getting deposits

You can remove keys that have not yet received the 32 ETH initial deposit.

> To remove them, go to the Keys section in the CMv2 widget and open the Remove tab.

Reviewer 2: is it free of charge?

#### Voluntary exit

Voluntary exits are appropriate when the protocol has requested an exit via the [Validators Exit Bus Oracle](https://docs.lido.fi/contracts/validators-exit-bus-oracle) to fulfill stETH withdrawal requests, or when you have insufficient bond to cover this key.

Please note this can't be done in the CMv2 widget, it can only be done through your validator client.

> Exiting validators outside of a protocol-requested exit is discouraged. If you plan to exit validators without a prior exit request, notify the Lido community in advance via the research forum.

Reviewer 2: community or CMC?

#### Ejection (triggerable withdrawals)

Through the CMv2 widget, you can force-exit an active validator directly from the Execution Layer using EIP-7002 triggerable withdrawals. This is an emergency measure. It is recommended to use voluntary exits broadcast via CL in normal operations.

:::info

> Read the [Lido Standard Node Operator Protocol (SNOP) for Validator Exits](https://github.com/lidofinance/documents-and-policies/blob/main/Lido%20on%20Ethereum%20Standard%20Node%20Operator%20Protocol%20-%20Validator%20Exits.md) before exiting any validator. It covers exit order, your obligations when processing requests, and what happens if you don't follow it.

Reviewer 2: I am not sure that this info should go under Ejection tbh, maybe consider moving under Exiting keys or somewhere else

:::

---

## Bond basics

Curated Module v2 introduces a bond for sub-NOs. Every new key requires a certain amount of bond (described in the section below), and it's used to cover certain losses from the NO. The amount depends on the Node Operator type and number of keys.

> Because the bond is staked, it generates staking rewards which can be claimed if they exceed the minimum required for the amount of keys submitted.

Reviewer 2: ```suggestion
Because the bond is staked (represented as stETH), it generates staking rewards which can be claimed if they exceed the minimum required for the amount of keys submitted.
```

:::info
These are Hoodi testnet confirmed values. Mainnet values will be confirmed before launch and may differ.
:::

---

### Locked bond

![Locked bond](/img/cm-guide/bond-locked.png)

When a **General Delayed Penalty** is reported against your operator, the penalty amount plus a fixed fee is immediately locked in `Accounting.sol`.

You can compensate the penalty from your excess bond. If needed, top up your bond first so there is enough excess to cover the penalty, then compensate it through the interface.

If no further action is taken by the Curated Module Committee, the Node Operator can do a transaction to unlock it after the lock period of 60 days ends.

:::warning
If you do not compensate a locked bond before the EasyTrack motion executes, those funds are **permanently burned**. In Phase 2, this will also result in a strike being assigned to the sub-Node Operator.
:::

---

### Penalty history

The CM interface also includes a penalty history section where you can review past penalties and actions recorded against your operator. For details on penalty types and how they are applied, see [Penalties](/run-on-lido/cm-v2/penalties).

---

### Unbonded keys

> If your bond falls below the required minimum, some of your keys become **Unbonded** and will not be deposited to, or will be requested to exit in case the deposited or active key is unbonded.

Reviewer 2: ```suggestion
If your bond falls below the required minimum, some of your keys become **Unbonded** and will not receive new deposits; if the key was already deposited or active, an exit will be requested.
```

---
sidebar_position: 6
---

# 🔐 Bond & Key Management

This page covers how to upload and manage your validator keys, and how bond works in CMv2. Bond is collateral you post to activate new validators from keys provided and back them. The amount required depends on your Node Operator type and number of keys.

---

## Key management

### Uploading keys

![Upload keys](/img/cm-guide/bond-upload-keys.png)

Before uploading keys, generate your validator keys and deposit data using a tool such as the [EthStaker Deposit CLI](https://github.com/eth-educators/ethstaker-deposit-cli) or your organization's standard key-generation process. Please make sure the generated keys use `0x02` withdrawal credentials and point to the [Lido Execution Layer Withdrawal Vault](https://etherscan.io/address/0xb9d7934878b5fb9610b3fe8a5e441e8fad7e293f).

> To upload the keys, go to the Keys tab in [cm.lido.fi](https://cm.lido.fi), and drag and drop the `deposit-data.json` file. The widget will tell you how much bond is required for the number of keys submitted.

Reviewer 3: There can be a mistake in the URL because this website is unreachable

<img width="688" height="353" alt="Image" src="https://github.com/user-attachments/assets/5bb91038-6c84-4381-8305-09c8589f20df" />
